template.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"io/ioutil"
+	"time"
+
+	"gopkg.in/yaml.v2"
+)
+
+type template struct {
```

Comment:
(something to consider for future) I wonder if we could reuse the `RequestOptions` somehow rather than duplicating the options